include/qengine_cpu.hpp

```diff
@@ -37,39 +36,40 @@ void rotate(BidirectionalIterator first, BidirectionalIterator middle, Bidirecti
  */
 class QEngineCPU : public QInterface, public ParallelFor {
 protected:
-    uint32_t randomSeed;
+    bool doNormalize;
     real1 runningNorm;
     complex* stateVec;
 
-    std::shared_ptr<std::default_random_engine> rand_generator;
-    std::uniform_real_distribution<real1> rand_distribution;
-
 public:
     QEngineCPU(bitLenInt qBitCount, bitCapInt initState, std::shared_ptr<std::default_random_engine> rgp = nullptr,
-        complex phaseFac = complex(-999.0, -999.0));
+        complex phaseFac = complex(-999.0, -999.0), bool partialInit = false);
+    QEngineCPU(QEngineCPUPtr toCopy)
+        : QInterface(toCopy->qubitCount, toCopy->rand_generator)
+    {
+        doNormalize = toCopy->doNormalize;
```

Comment:
Initialize in the constructor, for example:
```
    QEngineCPU(QEngineCPUPtr toCopy)
        : QInterface(toCopy->qubitCount, toCopy->rand_generator),
          doNormalize(toCopy->doNormalize)
    {
        CopyState(toCopy);
    }
```

Reply:
Done.